### PR TITLE
Sc api approvals

### PR DESCRIPTION
**Pull request by boblangley**
Connects to #1354

Resubmitted changes from #1401 / #1406 / #1418 with more commits separated by action, so the big ones (e.g. switching public types to private) can be more easily reviewed.

API approvals is implemented using the following philosophy:

- Message types are part of the public contract, especially the `Contracts`.
- SignalR uses message types for communication, so add those to the API approvals.
- Models used for Nancy modules should be part of the public contract
- All other classes should be internal and not part of API approvals.
- Nancy module action/paths should be part of API approvals.
- `RootModule` output should be part of API approvals.
- `ServiceControl.Transports` should be part of API approvals.

### CODE REVIEW

**Comment by mauroservienti** [comment]
@boblangley do I get it right that the only thing we're not testing when it comes to Nancy stuff are the incoming and outgoing HTTP headers?

**Comment by boblangley** [comment]
Yes, I forgot about that. I will start those today.

**Comment by boblangley** [comment]
Headers will require more effort than this PR will allow. I am going to open an issue for handling headers.

File: src/ServiceControl.AcceptanceTests/ServiceControl.AcceptanceTests.csproj.DotSettings [hunk 1]
```diff
@@ -0,0 +1,2 @@
+﻿<wpf:ResourceDictionary xml:space="preserve" xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml" xmlns:s="clr-namespace:System;assembly=mscorlib" xmlns:ss="urn:shemas-jetbrains-com:settings-storage-xaml" xmlns:wpf="http://schemas.microsoft.com/winfx/2006/xaml/presentation">
+	<s:String x:Key="/Default/CodeInspection/CSharpLanguageProject/LanguageLevel/@EntryValue">CSharp71</s:String></wpf:ResourceDictionary>
```
Comment (danielmarbach) [inline src/ServiceControl.AcceptanceTests/ServiceControl.AcceptanceTests.csproj.DotSettings]: Are these changes really necessary? I thought we are using the latest repo dotsettings from RepoStandards on the solution level. Shouldn't the root one be enough because those settings are inherited?

Comment (boblangley) [reply src/ServiceControl.AcceptanceTests/ServiceControl.AcceptanceTests.csproj.DotSettings]: Thanks @danielmarbach, good catch. I will move this to the solution level .dotsettings file.

**Comment by mauroservienti** [comment]
@Particular/servicecontrol-maintainers once @boblangley fixes the small glitch highlighted by @danielmarbach can we simply merge this one or there's a specific process to get changes in?

**Comment by danielmarbach** [comment]
@mauroservienti it seems someone specifically asked for @mikeminutillo to review. If that was the intention I'd say we wait for him. If that was not the intention I'm happy to merge as is and we can fix the dotsettings in another PR.

**Comment by mauroservienti** [comment]
I don't think there is any rush, @mikeminutillo will be back on Monday, I'm happy to wait for him.

**Comment by boblangley** [comment]
@danielmarbach @mikeminutillo is the assigned reviewer for the task force, that is why I specifically included him.

File: src/ServiceControl.Config/Xaml/Behaviours/RichTextBoxHelper.cs [hunk 2]
```diff
@@ -78,7 +78,8 @@ public static void SetDocumentXaml(DependencyObject obj, string value)
                         // When the document changes update the source
                         richTextBox.TextChanged += (obj2, e2) =>
                         {
-                            if (obj2 is RichTextBox richTextBox2)
+                            var richTextBox2 = (RichTextBox)obj2;
+                            if (richTextBox2 != null)
```
[the remark this answers is not in the document]
Comment (boblangley) [reply src/ServiceControl.Config/Xaml/Behaviours/RichTextBoxHelper.cs]: Code inspection. It reads better: 

> Is `obj2` a `RichTextBox`

Is easier to read than assigning it and checking for null, which is essentially answering the same question.

[the remark this answers is not in the document]
Comment (danielmarbach) [reply src/ServiceControl.Config/Xaml/Behaviours/RichTextBoxHelper.cs]: Hmm. that is weird because we switched that to the new code styles which allow pattern matching. The above code is the pattern matching variant which is recommended. We added that code when we move SC codebase to the new code styles. Could it be that you applied that recommendation with the custom dotsettings file per csproj?

[the remark this answers is not in the document]
Comment (bording) [reply src/ServiceControl.Config/Xaml/Behaviours/RichTextBoxHelper.cs]: Yeah, something seems off if it wanted you to move away from the newer C# style.
